src/python/pants/reporting/plaintext_reporter.py

```diff
@@ -182,6 +183,14 @@ def handle_output(self, workunit, label, s):
     self.flush()
 
   def emit(self, s, dest=ReporterDestination.OUT):
+    # In Py2, sys.stdout tries to coerce into ASCII, and will fail in coercing Unicode. So,
+    # we encode prematurely to handle unicode.
+    # In Py3, sys.stdout takes unicode, so will work normally.
+    #
+    # `self.settings.outfile` can also be `io.StringIO` instead of an std stream, in which case it only
+    # accepts unicode, so `s` does not need to be modified.
+    if PY2 and 'std' in str(self.settings.outfile):
```

Comment:
It would be good to loop back and see whether there is a more reliable way to do this... would you mind including a TODO in your next PR?